src/models.rs

```diff
@@ -991,20 +1042,32 @@ pub enum OperationResult {
     Failure { error: OperationError },
 }
 
-/// Represents the response of a batch operation.
+/// Batch file line JSON representation.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct BatchOperationResponse {
-    #[serde(rename = "@type")]
-    pub type_annotation: String,
-    pub inlined_responses: InlinedResponses,
+pub struct BatchFileItem {
+    /// Batch generatin request
+    pub request: GenerateContentRequest,
+    /// Batch request unique identifier
+    pub key: String,
 }
 
-/// Represents the output configuration of a batch operation.
+/// Represents the response of a batch operation.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct OutputConfig {
-    pub inlined_responses: InlinedResponses,
+#[serde(untagged)]
+pub enum BatchOperationResponse {
+    /// Response with inlined responses
+    InlinedResponses {
+        /// The inlined responses
+        #[serde(rename = "inlinedResponses")]
```

Comment:
I forgot about this interesting behavior of `serde` where it doesn't inherit the `rename_all` attribute when an enumeration is marked as `untagged`. Because of this, iterating through variants always returned `None`.